test/common/instance_api.test.lua

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env tarantool
+
+local socket = require('socket')
+local tap = require('tap')
+package.cpath = './?.so;' .. package.cpath
+local memcached = require('memcached')
+local test = tap.test('memcached instance api')
+
+local function is_port_open(port)
+    local sock, _ = socket.tcp_connect('127.0.0.1', port)
+    if sock == nil then
+        return false
+    end
+    return sock:sysconnect(0, port) and true or false
+end
+
+if type(box.cfg) == 'function' then
+    box.cfg{
+        wal_mode = 'none',
+        memtx_memory = 100 * 1024 * 1024,
+    }
+    box.schema.user.grant('guest', 'read,write,execute', 'universe')
+end
+
+test:plan(50)
+
+-- memcached.create()
+
+local mc_port = 11211
+local mc_name = 'memcached_xxx'
+local mc_space_name = 'memcached_aaa_space'
+
+local mc = memcached.create(mc_name, tostring(mc_port), {
+    space_name = mc_space_name
+})
+test:isnt(mc, nil, 'memcached instance object is not nil')
+test:is(is_port_open(mc_port), true, 'memcached instance is started')
+
+-- instance:stop()
+
+local self_instance = mc:stop()
+test:isnt(self_instance, nil, 'instance:stop() returns self object')
```

Comment:
I didn't quite understand from the README: isn't `self_instance == mc`?

Reply:
Yes, according to documented behavior in the README `mc` is the same as `self_instance`. assert checks that behavior.

Reply:
Wouldn't this kind of test be better?
```suggestion
test:is(self_instance, mc, 'instance:stop() returns self object')
```

Reply:
Fixed,